hackathon_site/hardware/test_api.py

```diff
@@ -136,6 +136,41 @@ def test_category_get_success(self):
 
         self.assertEqual(expected_response, data["results"][0])
 
+    def test_hardware_quantity_success(self):
+        self._login()
+
+        self.category = Category.objects.create(name="Microcontrollers", max_per_team=4)
+        self.hardware1 = Hardware.objects.create(name="Arduino", quantity_available=2)
+        self.hardware2 = Hardware.objects.create(name="ESP32", quantity_available=3)
+
+        self.hardware1.categories.add(self.category)
+        self.hardware2.categories.add(self.category)
+
+        expected_response = {
```

Comment:
As with the hardware API pr, we shouldn't hard-code the full response, since that's tested in the serializer test cases (and changing the serializer would berak this view unnecessarily).

Can do like you did in the other PR:

```python
expected_unique_hardware_counts = [0, 2]
actual_unique_hardware_counts = [result["unique_hardware_count"] for result in data["results"]]

self.assertEqual(expected_unique_hardware_counts, actual_unique_hardware_counts)
```